vm/src/obj/objbyteinner.rs

```diff
@@ -426,6 +426,69 @@ impl PyByteInner {
         }
     }
 
+    pub fn setitem(
+        &mut self,
+        needle: Either<PyIntRef, PySliceRef>,
+        object: PyObjectRef,
+        vm: &VirtualMachine,
+    ) -> PyResult {
+        match needle {
```

Comment:
Since we match on the two types of needle, maybe it makes sense to create two seperate functions to keep the function small in size. This is what I would take a look at, leaving the function as it is is also fine with me :).

Reply:
I agree, thanks for the tip :)